service/qos/service_level_controller.cc

```diff
@@ -125,7 +123,8 @@ future<> service_level_controller::stop() {
         // Abort source didn't fire, so do it now
         do_abort();
     }
-
+    
+    _global_controller_db->notifications_serializer.broken();
```

Comment:
This is followup to https://github.com/scylladb/scylladb/pull/18758#discussion_r1669802920 conversation, since I've dropped that commit.

I don't think the semaphore is used to (2) control an access to service levels. 
To me it looks like it's only used to serialize write access to sl cache (as the name suggests). Nowhere in the code methods which uses the semaphore are in try/catch blocks and I also didn't find any mentions that the semaphore control the access to service levels.

So I've moved semaphore `broken()` to the `stop()` method, which should fix the issue with crash in `topology_state_load()`.

Reply:
Ok, so why do we break this semaphore at all? Why was this call introduced in the first place? Maybe we can just remove `.broken()` altogether?

I'd rather not remove it without understanding why it was introduced in the first place because of the [Chesterton's fence](https://en.wikipedia.org/wiki/G._K._Chesterton#Chesterton's_fence) principle.

We can handle this in a followup, though.